subsys/bluetooth/controller/ll_sw/ull_scan.c

```diff
@@ -411,17 +418,81 @@ uint8_t ull_scan_enable(struct ll_scan_set *scan)
 			scan->ull.ticks_slot = 0U;
 		} else {
 			scan->ull.ticks_slot = ticks_interval -
-				HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_XTAL_US);
+					       ticks_slot_overhead;
 		}
 
 		lll->ticks_window = 0U;
 	}
 
-	if (IS_ENABLED(CONFIG_BT_CTLR_LOW_LAT)) {
-		ticks_slot_overhead = MAX(scan->ull.ticks_active_to_start,
-					  scan->ull.ticks_prepare_to_start);
+	handle = ull_scan_handle_get(scan);
+
+	if (false) {
+
+#if defined(CONFIG_BT_CTLR_ADV_EXT) && defined(CONFIG_BT_CTLR_PHY_CODED)
+	} else if (handle == SCAN_HANDLE_1M) {
+		const struct ll_scan_set *scan_coded;
+
+		scan_coded = ull_scan_set_get(SCAN_HANDLE_PHY_CODED);
+		if (IS_PHY_ENABLED(scan_coded, PHY_CODED) &&
+		    (lll->ticks_window != 0U)) {
+			uint32_t ticks_interval_coded;
+
+			ticks_interval_coded = HAL_TICKER_US_TO_TICKS(
+				(uint64_t)scan_coded->lll.interval *
+				SCAN_INT_UNIT_US);
+			if ((ticks_interval + lll->ticks_window) ==
+			    (ticks_interval_coded +
+			     scan_coded->lll.ticks_window)) {
+				if (IS_ENABLED(CONFIG_BT_CTLR_SCAN_UNRESERVED)) {
+					scan->ull.ticks_slot = 0U;
+				} else {
+					scan->ull.ticks_slot =
+						lll->ticks_window -
+						ticks_slot_overhead -
+						HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_START_US) -
+						HAL_TICKER_US_TO_TICKS(EVENT_TICKER_RES_MARGIN_US);
+				}
+			}
+		}
+
+		ticks_offset = 0U;
+	} else if (handle == SCAN_HANDLE_PHY_CODED) {
+		struct ll_scan_set *scan_1m;
+
+		scan_1m = ull_scan_set_get(SCAN_HANDLE_1M);
+		if (IS_PHY_ENABLED(scan_1m, PHY_1M) &&
+		    (lll->ticks_window != 0U)) {
+			uint32_t ticks_interval_1m;
+
+			ticks_interval_1m = HAL_TICKER_US_TO_TICKS(
+				(uint64_t)scan_1m->lll.interval *
+				SCAN_INT_UNIT_US);
+			if ((ticks_interval + lll->ticks_window) ==
+			    (ticks_interval_1m + scan_1m->lll.ticks_window)) {
```

Comment:
Commit message says `same interval and window duration that is half the interval was requested`. I see that there is a check if both interval and window are equal. Where is this "half the interval time"?

Reply:
Fixed the code to compare the sum of the window duration to be equal to their same scan interval. This way one scan window follows after the other and the window is complete by the next scan interval.